catalog/app/containers/Bucket/FileProperties.tsx

```diff
@@ -0,0 +1,121 @@
+
+import * as requests from './requests'
+
+interface FilePropertyProps extends M.TypographyProps {
+  children: React.ReactNode
+  className: string
+}
+
+const useFilePropertyStyles = M.makeStyles((t) => ({
+  root: {
+    alignItems: 'center',
+    display: 'inline-flex',
+  },
+  icon: {
+    color: t.palette.text.hint,
+    fontSize: 16,
+    marginRight: '2px',
+  },
+}))
+
+function FileProperty({ className, children, ...props }: FilePropertyProps) {
+  const classes = useFilePropertyStyles()
+  return (
+    <M.Typography
+      className={cx(classes.root, className)}
+      component="span"
+      variant="body2"
+      {...props}
+    >
+      {children}
+    </M.Typography>
+  )
+}
+
+function FilePropertiesSkeleton({ className }: { className: string }) {
+  const classes = useFilePropertiesStyles()
+  return (
+    <div className={cx(classes.root, className)}>
+      <Skeleton animate className={classes.property} width="80px" height="14px" />
+      <Skeleton animate className={classes.property} width="80px" height="14px" />
+    </div>
+  )
+}
+
+interface FilePropertiesProps {
+  data: $TSFixMe
+  className: string
+}
+
+const useFilePropertiesStyles = M.makeStyles((t) => ({
+  root: {
+    alignItems: 'center',
+    display: 'flex',
+  },
+  property: {
+    marginLeft: t.spacing(1),
+  },
+}))
+
+interface FilePropertiesBareProps {
+  className?: string
+  lastModified?: Date
+  size?: number
+}
+
+function FileProperties({ className, lastModified, size }: FilePropertiesBareProps) {
+  const classes = useFilePropertiesStyles()
+
+  const today = React.useMemo(() => new Date(), [])
+  const formattedDate = React.useMemo(
+    () =>
+      lastModified
+        ? dateFns.format(
+            lastModified,
+            today.getFullYear() === lastModified.getFullYear() ? 'd MMM' : 'd MMM yyyy',
+          )
+        : null,
+    [lastModified, today],
+  )
+  const formattedSize = React.useMemo(() => (size ? readableBytes(size) : null), [size])
+
+  return (
+    <div className={cx(classes.root, className)}>
+      {formattedDate && (
+        <FileProperty title={lastModified?.toLocaleString()} className={classes.property}>
+          {formattedDate}
+        </FileProperty>
+      )}
+      {formattedSize && (
+        <FileProperty className={classes.property}>{formattedSize}</FileProperty>
+      )}
+    </div>
+  )
+}
+
+export const Container = FileProperties
+
+export function Wrapper({ className, data }: FilePropertiesProps) {
+  const classes = useFilePropertiesStyles()
+  return (
+    <div className={cx(classes.root, className)}>
+      {data.case({
+        Ok: requests.ObjectExistence.case({
+          Exists: ({ lastModified, size }: { lastModified?: Date; size?: number }) => (
+            <FileProperties {...{ className, lastModified, size }} />
+          ),
+          _: () => <FilePropertiesSkeleton className={className} />,
+        }),
+        Err: (e: Error) => e.message,
```

Comment:
this may look a little awkward in the ui depending on the message (and it also does not have any styling). maybe show an error icon instead and display error message as a tooltip?

Reply:
Thanks, I didn't mean it, I forgot to change